content/blog/2019/egress-traffic-control-in-istio-part-2/index.md

```diff
@@ -0,0 +1,101 @@
+If the application sends HTTP requests and the egress gateway performs TLS origination, you can monitor HTTP
+information, e.g. HTTP methods, headers and URL paths, and you can
+[define policies](/blog/2018/egress-monitoring-access-control) based on the HTTP information. If the application
+performs TLS origination, for TLS traffic you can
+[monitor SNI and the service account](/docs/tasks/traffic-management/egress/egress_sni_monitoring_and_policies/) of the
+source pod, and define policies based on them.
+
+The following diagram shows Istio's security architecture, augmented with L3 firewall (part of the
+[additional security mechanisms](/docs/tasks/traffic-management/egress/egress-gateway/#additional-security-considerations)
+provided outside of Istio by the cluster/cloud provider).
+The L3 firewall can have a trivial configuration that would allow incoming traffic into Istio ingress gateway pods only and
+outgoing traffic from Istio egress gateway pods only. Note that the Istio proxy of the egress gateway performs
+policy enforcement and reporting in the same way as the sidecar proxies in the application pods.
+
+{{< image width="80%" link="./SecurityArchitectureWithL3Firewalls.svg" caption="Istio Security Architecture with Egress Gateway and L3 Firewall" >}}
+
+Now let's examine possible attacks and let me show you how the secure control of egress traffic in Istio prevents them.
+
+## Possible attacks and their prevention
+
+Consider the following security policy with regard to egress traffic:
+
+1. Application A is allowed to access `*.ibm.com` (all the external services with URL matching `*.ibm.com`,
+  e.g. `www.ibm.com`)
+1. Application B is allowed to access `mongo1.composedb.com`
+1. All egress traffic must be monitored
+
+Now consider a scenario in which one of application A's pods is compromised. Suppose the attackers have the
+following goals:
+
+1. Application A will try to access `*.ibm.com` unmonitored
+1. Application A will try to access `mongo1.composedb.com`
+
+Note that application A is allowed to access `*.ibm.com`, so the attacker is able to access it. There is no way
+to prevent such access since there is no way to distinguish, at least initially, between the original and the
+compromised versions of application A. However, you want to monitor any access to external services to be able to
+detect suspicious traffic, for example by applying anomaly detection tools on logs of the egress traffic.
+The attackers, on the contrary, want to access external services unmonitored, so the attack will not be detected.
+The second goal of the attackers is to access `mongo1.composedb.com`, which is forbidden for application A. Istio
+must enforce the policy that forbids access of application A to `mongo1.composedb.com` and must prevent the attack.
+
+Now let's see which attacks the attackers will try to perform to achieve their goals and how Istio secure egress traffic
+control will prevent each kind of attack. The attackers may try to:
```

Comment:
@rcaballeromx  I do not like this change:

1. Istio does not help to prevent, it actually prevents
2. Common attacks - I do not claim that these are common attacks, these are the attacks that came to my mind.